cudapoa/tests/Test_CudapoaNW.cu

```diff
@@ -304,6 +305,220 @@ TEST_P(NWTest, TestNWCorrectness)
+                             incoming_edges,
+                             outgoing_edge_count,
+                             read,
+                             read_count,
+                             scores,
+                             traces,
+                             batch_size.matrix_sequence_dimension,
+                             batch_size.max_nodes_per_graph,
+                             alignment_graph,
+                             alignment_read,
+                             batch_size.alignment_band_width,
+                             batch_size.max_banded_pred_distance,
+                             gap_score,
+                             mismatch_score,
+                             match_score,
+                             aligned_nodes,
+                             adaptive);
+    }
+    else
+    {
+        runNWbanded<SizeT>(nodes,
+                           graph,
+                           node_id_to_pos,
+                           graph_count,
+                           incoming_edge_count,
+                           incoming_edges,
+                           outgoing_edge_count,
+                           read,
+                           read_count,
+                           scores,
+                           batch_size.matrix_sequence_dimension,
+                           batch_size.max_nodes_per_graph,
+                           alignment_graph,
+                           alignment_read,
+                           batch_size.alignment_band_width,
+                           gap_score,
+                           mismatch_score,
+                           match_score,
+                           aligned_nodes,
+                           adaptive);
+    }
+
+    GW_CU_CHECK_ERR(cudaDeviceSynchronize());
+
+    //input and output buffers are the same ones in unified memory, so the results are updated in place
+    //results are stored in alignment_graph and alignment_read; return string representation of those
+    auto res = std::make_pair(genomeworks::stringutils::array_to_string<SizeT>(alignment_graph, *aligned_nodes, ","),
+                              genomeworks::stringutils::array_to_string<SizeT>(alignment_read, *aligned_nodes, ","));
+
+    GW_CU_CHECK_ERR(cudaFree(nodes));
+    GW_CU_CHECK_ERR(cudaFree(graph));
+    GW_CU_CHECK_ERR(cudaFree(node_id_to_pos));
+    GW_CU_CHECK_ERR(cudaFree(incoming_edges));
+    GW_CU_CHECK_ERR(cudaFree(incoming_edge_count));
+    GW_CU_CHECK_ERR(cudaFree(outgoing_edges));
+    GW_CU_CHECK_ERR(cudaFree(outgoing_edge_count));
+    GW_CU_CHECK_ERR(cudaFree(scores));
+    GW_CU_CHECK_ERR(cudaFree(alignment_graph));
+    GW_CU_CHECK_ERR(cudaFree(read));
+    GW_CU_CHECK_ERR(cudaFree(alignment_read));
+    GW_CU_CHECK_ERR(cudaFree(aligned_nodes));
+    if (traceback)
+    {
+        GW_CU_CHECK_ERR(cudaFree(traces));
+    }
+
+    return res;
+}
+
+class NWbandedTest : public ::testing::Test
+{
+public:
+    BasicNW* nw;
```

Comment:
Could you use a `std::unique_ptr<BasicNW>` here? Then you should be able to drop the `TearDown()` function.

Reply:
done.